packages/babel-preset-expo/src/expo-inline-manifest-plugin.ts

```diff
@@ -157,41 +161,50 @@ export function expoInlineManifestPlugin(api: ConfigAPI & typeof import('@babel/
   const platform = api.caller(getPlatform);
   const possibleProjectRoot = api.caller(getPossibleProjectRoot);
   const shouldInline = platform === 'web' || isReactServer;
+
+  // Early exit: return a no-op plugin if we're not going to inline anything
+  if (!shouldInline) {
+    return {
+      name: 'expo-inline-manifest-plugin',
+      visitor: {},
+    };
+  }
+
   return {
     name: 'expo-inline-manifest-plugin',
-    visitor: {
-      MemberExpression(path: any, state: any) {
-        // Web-only/React Server-only feature: the native manifest is provided dynamically by the client.
-        if (!shouldInline) {
-          return;
-        }
 
-        if (
-          !t.isIdentifier(path.node.object, { name: 'process' }) ||
-          !t.isIdentifier(path.node.property, { name: 'env' })
-        ) {
-          return;
-        }
+    pre() {
+      (this as InlineManifestState).projectRoot = possibleProjectRoot || this.file.opts.root || '';
```

Comment:
nit: same as before, we can define `PluginObj<T>` for the `this` type

other nit: we can probably hoist `this.file.opts.root || ''` up since `possibleProjectRoot` is otherwise separated from here